### PR TITLE
QueryOffset and QuerySequence

### PR DESCRIPTION
**Pull request by simone-fariselli**
Add QueryOffsetWithDefaultValue
Add tests
Change in QueryOffset and QuerySequence: on reference null or empty throws as discussed with @Gsantomaggio 
Refactor

### CODE REVIEW

File: RabbitMQ.Stream.Client/StreamSystem.cs [hunk 1]
```diff
@@ -111,6 +111,18 @@ public async Task<bool> StreamExists(string stream)
             return response.StreamInfos is { Count: >= 1 } &&
                    response.StreamInfos[stream].ResponseCode == ResponseCode.Ok;
         }
+        private static void MaybeThrowQueryException(string reference, string stream)
+        {
+            if (string.IsNullOrEmpty(reference) || string.IsNullOrEmpty(stream))
+            {
+                throw new QueryException("Stream name can't be empty");
```
Comment (Gsantomaggio) [inline RabbitMQ.Stream.Client/StreamSystem.cs:118]: I would change in:
```csharp
QueryException("Stream name reference  and can't be empty");
````

File: RabbitMQ.Stream.Client/StreamSystem.cs [hunk 2]
```diff
@@ -121,17 +133,32 @@ public async Task<bool> StreamExists(string stream)
...
+
+        /// <summary>
+        /// QueryOffsetWithDefaultValue retrieves the last consumer offset stored
+        /// given a consumer name and stream name 
+        /// or 0 as default value if thre is no stored value
+        /// </summary>
+        /// <param name="reference">Consumer name</param>
+        /// <param name="stream">Stream name</param>
+        /// <returns></returns>
+        public async Task<ulong> QueryOffsetWithDefaultValue(string reference, string stream)
```
Comment (Gsantomaggio) [inline RabbitMQ.Stream.Client/StreamSystem.cs:152]: I would add an overload for `QueryOffset(string reference, string stream, ulong defaultValue)`

**Review by Gsantomaggio — CHANGES_REQUESTED** [review]

File: RabbitMQ.Stream.Client/StreamSystem.cs (hunk 1 above)
Comment (simone-fariselli) [reply RabbitMQ.Stream.Client/StreamSystem.cs:118]: Ok, 
I will set the message "Stream name and reference can't be empty or null".

**Comment by simone-fariselli** [comment]
Removed QueryOffsetWithDefaultValue as discussed with @Gsantomaggio

**Review by Gsantomaggio — APPROVED** [review]

**Comment by Gsantomaggio** [comment]
thank you @simone-fariselli

**Pull request merged**